### PR TITLE
docs(tree2): Add docs and fix typos

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Fixes some typos, and adds/improves documentation on some APIs

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

**Review by chentong7 — APPROVED** [review]

File: api-report/tree2.api.md [hunk 1]
```diff
@@ -465,13 +465,9 @@ export interface DeltaVisitor {
     create(content: Delta.ProtoNodes, destination: FieldKey): void;
     destroy(detachedField: FieldKey, count: number): void;
     detach(source: Range_2, destination: FieldKey): void;
-    // (undocumented)
```
Comment (Josmithr) [inline api-report/tree2.api.md:468]: Woo! Thanks!

File: experimental/dds/tree2/src/core/tree/anchorSet.ts [hunk 2]
```diff
@@ -569,7 +569,14 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents>, AnchorLoca
...
 	/**
-	 * Updates the anchors according to the changes described in the given delta
+	 * Provides a visitor that can be used to mutate this {@link AnchorSet}.
+	 *
+	 * @returns A visitor that can be used to mutate this {@link AnchorSet}.
+	 *
+	 * @remarks
+	 * Mutating the {@link AnchorSet} does NOT update the forest.
+	 * The visitor must be released after use.
+	 * It is invalid to acquire a visitor without releasing the previous one.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/core/tree/anchorSet.ts:579]: Does doing so result in an error being thrown? Or just undefined behavior?

File: experimental/dds/tree2/src/core/tree/visitDelta.ts [hunk 3]
```diff
@@ -180,9 +180,52 @@ export interface DeltaVisitor {
 	 */
 	replace(newContentSource: FieldKey, range: Range, oldContentDestination: FieldKey): void;
 
+	/**
+	 * Tells the visitor's "owner" that it should update its "current location" to be the Node at the specified index
+	 * within the Field that is the current "current location".
+	 * Future calls to methods of {@link DeltaVisitor} will assume that's the location where their effects are happening.
+	 * @param index - The index (within the Field) of the node that should become the new "current location".
+	 *
+	 * @remarks This should only be called when the "current location" is a Field.
```
Comment (Josmithr) [inline experimental/dds/tree2/src/core/tree/visitDelta.ts:189]: What is the behavior if it isn't currently a field? Do we throw?

**Review by Josmithr — APPROVED** [review]
Thanks for adding the docs! My only feedback is that it would be good to document our failure mode contracts for APIs that have preconditions.

File: experimental/dds/tree2/src/core/tree/visitDelta.ts [hunk 4]
```diff
@@ -180,9 +180,52 @@ export interface DeltaVisitor {
...
+	 *
+	 * @remarks This should only be called when the "current location" is a Node.
+	 */
 	enterField(key: FieldKey): void;
+
+	/**
+	 * Tells the visitor's "owner" that it should update its "current location" to be the Node which contains the Field
+	 * that is the current "current location".
+	 * Future calls to methods of {@link DeltaVisitor} will assume that's the location where their effects are happening.
+	 * @param key - NOT USED.
```
Comment (yann-achard-MS) [inline experimental/dds/tree2/src/core/tree/visitDelta.ts:225]:
```suggestion
	 * @param key - The key of the field that is being exited.
```

File: experimental/dds/tree2/src/core/tree/visitDelta.ts [hunk 5]
```diff
@@ -180,9 +180,52 @@ export interface DeltaVisitor {
 	 */
 	replace(newContentSource: FieldKey, range: Range, oldContentDestination: FieldKey): void;
 
+	/**
+	 * Tells the visitor's "owner" that it should update its "current location" to be the Node at the specified index
```
[the remark this answers is not in the document]
Comment (alexvy86) [reply experimental/dds/tree2/src/core/tree/visitDelta.ts:188]: > I'm curious what motivated this reference to an owner
My lack of familiarity with tree 😄 . I'll update it.

**Pull request merged**